### PR TITLE
Nnnn refining property wrapper related initialization patch 1

### PR DESCRIPTION
**Pull request by amritpan**
Hey Filip, Freddy,

I see that you made quick work of the pitch draft today! In this pr, I've added an alternative intro option for the pitch. 

I also think it would be great to put the init examples that were presented in the forums.swift discussion thread for this pitch in our motivation section. Freddy, the breadth of examples you offered in the Placeholder Type pitch helped me get started in reading proposals so I'm thinking of that here. I personally have difficulty understanding concepts until I see or write some code and similarly, I only began to understand the issues with the memberwise init rules as they stand when I saw the example code in the discussion. However, I do understand if they are too verbose, and in which case, perhaps we can just omit the entirety and just add a direct link to Holly's comment with the examples.

### CODE REVIEW

File: proposals/NNNN-refining-property-wrapper-related-initialization.md [hunk 1]
```diff
@@ -7,7 +7,7 @@
...
+[SE 0258](https://github.com/apple/swift-evolution/blob/master/proposals/0258-property-wrappers.md) introduced property wrappers and [SE 0293](https://github.com/apple/swift-evolution/blob/main/proposals/0293-extend-property-wrappers-to-function-and-closure-parameters.md#detailed-design) expanded them with function-like declarations. Today, property wrapper initialization exhibits inconsistencies due to its growing versatility. Specifically, memberwise initializers use complex, poorly documented rules and projection initialization remains limited. This proposal will simplify synthesized memberwise initialization for types with wrapped properties and extend projection value initialization to include global, type, and local wrapped properties.
```
Comment (filip-sakel) [inline proposals/NNNN-refining-property-wrapper-related-initialization.md:10]:
```suggestion
[SE 0258](https://github.com/apple/swift-evolution/blob/master/proposals/0258-property-wrappers.md) introduced property wrappers and [SE 0293](https://github.com/apple/swift-evolution/blob/main/proposals/0293-extend-property-wrappers-to-function-and-closure-parameters.md#detailed-design) expanded them with function-like declarations. Today, property wrapper initialization exhibits inconsistencies due to its growing versatility. Specifically, memberwise initializers use complex, poorly documented rules and projection initialization remains limited. This proposal will simplify synthesized memberwise initialization for types with wrapped properties and extend projection value initialization to global, type, and local wrapped properties.
```

File: proposals/NNNN-refining-property-wrapper-related-initialization.md [hunk 2]
```diff
@@ -19,6 +19,53 @@ Today, the rules for how wrapped properties are represented in the synthesized m
...
+  @WrapperWithDefaultArgs var value5: Int
+}
+
+let s = S(value1: WrapperWithDefaultInit(wrappedValue: 1),
+          value2: 2,
+          value3: WrapperWithArgs(wrappedValue: 3, extraArg: true),
+          value4: 4,
+          value5: 5)
+```
+
```
Comment (filip-sakel) [inline proposals/NNNN-refining-property-wrapper-related-initialization.md:54]: Thanks for adding Holly's example! I propose a slightly simplified version:

```swift
@propertyWrapper
struct Wrapper {
    let wrappedValue: Int
    
    init(wrappedValue: Int = 5) {
        self.wrappedValue = wrappedValue
    }
}

@propertyWrapper
struct ArgumentWrapper {
    let wrappedValue: Int
    let arg: Int
}

struct Client {
    @Wrapper var a
    @Wrapper var b = 2
    @ArgumentWrapper var c: Int
    @ArgumentWrapper(arg: 0) var d = 17
}

let client = Client(
    a: Wrapper(wrappedValue: 1),
    b: 2,
    c: ArgumentWrapper(wrappedValue: 3, arg: 0),
    d: 4
)
```

Compromises: With two wrapper types we can't have the `value5` example. However, I don't think we need to; the problem is demonstrated either way.

Thoughts?

**Review by filip-sakel — APPROVED** [review]
Thanks for adding the proposed features to the introduction! Also, Holly's example makes the proposal more graspable. I just wrote a more succinct version; let me know what you think. Lastly, I agree it's important to include examples, so thanks for bringing that up!

File: proposals/NNNN-refining-property-wrapper-related-initialization.md (hunk 2 above)
Comment (amritpan) [reply proposals/NNNN-refining-property-wrapper-related-initialization.md:54]: I agree! And this version looks great - even more readable than the original.

Comment (filip-sakel) [reply proposals/NNNN-refining-property-wrapper-related-initialization.md:54]: Great! Feel free to commit the changes and merge.

**Pull request merged**